js/wp-seo-wordpress-scraper.js

```diff
@@ -0,0 +1,394 @@
+    if(this.analyzerDataQueue.length > 0){
+        var currentData = this.analyzerDataQueue.shift();
+        this.replaceVariables(this.analyzerData[currentData], currentData, this.formattedData);
+    }else{
+        if(typeof this.refObj.snippetPreview === "undefined") {
+            this.refObj.init();
+        }else{
+            this.refObj.reloadSnippetText();
+        }
+        this.refObj.runAnalyzerCallback();
+    }
+};
+
+/**
+ * gets content from the content field, if tinyMCE is initialized, use the getContent function to get the data from tinyMCE
+ * @returns textString
+ */
+YoastSEO_WordPressScraper.prototype.getContentTinyMCE = function() {
+    var val = document.getElementById( "content").value;
+    if(tinyMCE.editors.length !== 0){
+        val = tinyMCE.get( "content").getContent();
+    }
+    return val;
+};
+
+/**
+ * gets data from hidden input fields. Is triggered on click in the snippet preview. Fetches data and inserts into snippetPreview
+ * @param inputType
+ */
+YoastSEO_WordPressScraper.prototype.getInputFieldsData = function ( ev ) {
+    var inputType = ev.currentTarget.id.replace(/snippet_/i, "");
+    switch( inputType ){
+        case "title":
+            document.getElementById( "snippet_title" ).textContent = document.getElementById( "yoast_wpseo_title" ).value;
+            break;
+        case "meta":
+            document.getElementById( "snippet_meta" ).textContent = document.getElementById( "yoast_wpseo_metadesc" ).value;
+            break;
+        case "url":
+            //var urlBase = document.getElementById("sample-permalink").textContent.replace(/https?:\/\//i, "").split("/")[0]+"/";
+            var newUrl = document.getElementById( "snippet_cite" ).textContent;
+            //newUrl = newUrl.replace(urlBase, "");
+            document.getElementById("editable-post-name").textContent = newUrl;
+            //document.getElementById("editable-post-name-full").textContent = newUrl;
+            break;
+        default:
+            break;
+    }
+};
+
+/**
+ * Replaces %% strings with WordPress variables
+ * @param textString
+ * @param type
+ * @param object
+ * @param target
+ * @returns {string}
+ */
+YoastSEO_WordPressScraper.prototype.replaceVariables = function( textString, type, object) {
+    if(typeof textString === "undefined"){
+        object[type] = "";
+        this.runDataQueue();
+
+    } else {
+        textString = this.titleReplace(textString);
+        textString = this.defaultReplace(textString);
+        textString = this.parentReplace(textString);
+        textString = this.doubleSepReplace(textString);
+        textString = this.excerptReplace(textString);
+
+        if (textString.indexOf("%%") !== -1 && textString.match(/%%[a-z0-9_-]+%%/i) !== null && typeof this.replacedVars !== "undefined") {
+            var regex = /%%[a-z0-9_-]+%%/gi;
+            var matches = textString.match(regex);
+            for (var i = 0; i < matches.length; i++) {
```

Comment:
I would move the replace logic to a different function, like `replaceVariable`. Just like you have the title, default, parent, double and excerpt replacements.